parquet-column/src/main/java/org/apache/parquet/internal/column/columnindex/ColumnIndexBuilder.java

```diff
@@ -543,6 +546,11 @@ public static ColumnIndex build(
    *          the statistics to be added
    */
   public void add(Statistics<?> stats) {
+    if (stats.isEmpty()) {
```

Comment:
It should be invalid if the stats are empty. Previously we set it as a null page. @gszadovszky please correct me if I am wrong.

Reply:
I am not sure how column index writing worked for the "rewriter". This rewriting thing is newer than the column index. In the normal writing scenario `stats` cannot be empty since we are just creating these objects during the write path.

Reply:
This is especially used when the `ColumnIndex` is null during rewriting. And we pass empty statistics to the `ColumnIndexBuilder` to avoid NPE.

Reply:
Let me try to understand what happens here. `convertStatistics` is used to recover page statistics from ColumnIndex or original page header if the ColumnIndex is unavailable. The problem emerges when ColumnIndex is unavailable. Am I correct? If true, then why do we need those changes in the ColumnIndexBuilder?

Reply:
The problem happens when both the `ColumnIndex` and the page header `Statistics` are null. Because we get `null` returned from the `convertStatistics`. However, the `ParquetFileWriter.writeDataPage` needs the page statistics. So here we pass invalid page statistics to avoid the NPE and overwrite the column statistics in the end. Otherwise, we need to add some methods that don't need page statistics.

Reply:
From what you have said, it seems that the problem comes from the input file which has valid aggregate statistics for the column chunk but does not write page statistics in the page header. Should we just fix the NPE in the page header and leave other parts as is?

Reply:
> Should we just fix the NPE in the page header and leave other parts as is?

Updated the implementation.